llvm/lib/Transforms/Yk/ControlPoint.cpp

```diff
@@ -121,8 +122,24 @@ class YkControlPoint : public ModulePass {
     // Get function containing the control point.
     Function *Caller = OldCtrlPointCall->getFunction();
 
-    // Find all live variables just before the call to the control point.
+    // Check that the control point was called inside, what appears to be, an
+    // interpreter loop, i.e. in a top-level loop in some function.
     DominatorTree DT(*Caller);
+    const LoopInfo Loops(DT);
+    bool FoundInLoop = false;
```

Comment:
Is this equivalent to something like `Loops.any_of(..)` (which I think might be the C++ equivalent of Rust's `any` in an iterator)?

Reply:
Fixed in bd256da